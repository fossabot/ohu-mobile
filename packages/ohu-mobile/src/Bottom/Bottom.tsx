import { defineComponent } from '../_utils/defineComponent';
import { BottomProps } from './types';


const Bottom = defineComponent<BottomProps>('bottom').create({
  render() {
    const { $slots } = this;
    return (
      <div class={this.root()}>{$slots.default}</div>
    );
  }
});

export default Bottom;
