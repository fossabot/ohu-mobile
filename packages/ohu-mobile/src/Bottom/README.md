# Utils: Bottom

## Explain

In this version, it implement by css, So you must add `viewport-fit=cover` in meta.

## Usage

```js
import { Bottom } from '@ohu-mobile/core';
```


