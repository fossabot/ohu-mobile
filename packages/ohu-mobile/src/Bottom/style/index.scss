@import '../../_styles/variables.scss';
@import '../../_styles/mixins.scss';

$bottomPrefixClass: #{$prefix}bottom;

.#{$bottomPrefixClass} {
  width: 100%;
  position: fixed;
  bottom: 0;
  left: 0;
  padding-bottom: constant(safe-area-inset-bottom); /* iOS < 11.2 */
  padding-bottom: env(safe-area-inset-bottom); /* iOS >= 11.2 */
}
