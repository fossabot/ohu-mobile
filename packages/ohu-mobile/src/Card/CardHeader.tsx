import { componentFactory } from 'vue-tsx-support';
import props from 'vue-strict-prop';
import { $prefix } from '../_config/variables';

export const cardHeaderBaseName = `${$prefix}card-header`;
const cardHeaderExtraCls = `${$prefix}card-header__extra`;
const CardHeader = componentFactory.create({
  name: cardHeaderBaseName,
  props: {
    status: props.ofStringLiterals('error', 'success', 'normal').default('normal'),
    bold: props(Boolean).default(false),
    extra: String,
  },
  computed: {
    cls() {
      return {
        [cardHeaderBaseName]: true,
        [`is-${this.status}`]: true,
        'is-bold': this.bold,
      };
    },
  },
  render() {
    const { $slots, extra } = this;
    const extraContent = $slots.extra || extra;
    return (
      <div class={this.cls}>
        { $slots.default }
        { extraContent && <div class={cardHeaderExtraCls}>{extraContent}</div> }
      </div>
    );
  },
});

export default CardHeader;
