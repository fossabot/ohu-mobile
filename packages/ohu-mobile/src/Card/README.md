
# Usage

```js
import { Card } from 'ohu-mobile';
```


