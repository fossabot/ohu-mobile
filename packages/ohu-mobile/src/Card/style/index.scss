@import '../../_styles/variables.scss';
@import '../../_styles/mixins.scss';
$cardPrefixClass: #{$prefix}card;
$cardHeaderPrefixClass: #{$cardPrefixClass}-header;
$cardHeaderTopOffset: 22px;
$cardHeaderLeftOffset: 24px;

.#{$cardPrefixClass} {
  background: $cardBackground;
  &__content {
    padding: 24px;
  }
  &.has-radius {
    border-radius: 8px;
  }
  &.has-shadow {
    box-shadow: $cardBoxShadow;
  }
}

.#{$cardHeaderPrefixClass} {
  position: relative;
  padding: $cardHeaderTopOffset $cardHeaderLeftOffset;
  font-size: 32px;
  line-height: 1.375em;
  color: $colorTextBase;
  font-weight: 400;
  &__extra {
    @include center(y);
    right: $cardHeaderLeftOffset;
    color: $colorTextMinor;
    font-size: 32px;
  }
  &.is-bold {
    font-size: 36px;
    line-height: 1.38888889em;
    font-weight: bold;
  }
  // ================== state ==================
  &.is-error {
    .#{$cardHeaderPrefixClass}__extra {
      color: $colorTextError;
    }
  }
  &.is-success {
    .#{$cardHeaderPrefixClass}__extra {
      color: $colorTextHighlight;
    }
  }
}
