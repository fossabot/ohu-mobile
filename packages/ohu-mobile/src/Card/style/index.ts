import '../../_styles';
import './index.scss';

// style dependencies
import '../../Divider/style';
