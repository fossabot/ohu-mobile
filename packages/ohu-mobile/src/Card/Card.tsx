import { componentFactory } from 'vue-tsx-support';
import props from 'vue-strict-prop';
import { getVNodesByName } from '../_utils/vnode';
import { cardHeaderBaseName } from './CardHeader';
import { $prefix } from '../_config/variables';
import Divider from '../Divider';

const cardBaseName = `${$prefix}card`;
const cardContentCls = `${cardBaseName}__content`;
const Card = componentFactory.create({
  name: cardBaseName,
  props: {
    shadow: props(Boolean).default(false),
    borderless: props(Boolean).default(false),
    divider: props(Boolean).default(false),
  },
  computed: {
    cls() {
      return {
        [cardBaseName]: true,
        'has-shadow': !this.borderless && this.shadow,
        'has-radius': !this.borderless,
      };
    },
  },
  render() {
    const { cls, $slots, divider } = this;
    // detect card-header, extra other vnode as content
    if ($slots.default instanceof Array) {
      const headerNodes = getVNodesByName($slots.default, cardHeaderBaseName);
      if (headerNodes.length > 0) {
        const contentNodes = getVNodesByName($slots.default, (name) => {
          return name !== cardHeaderBaseName;
        });
        return (
          <div class={cls}>
            {headerNodes}
            { divider && <Divider></Divider> }
            {
              contentNodes.length > 0
              &&
              <div class={cardContentCls}>
                { contentNodes }
              </div>
            }
          </div>
        );
      }
    }
    return (
      <div class={cls}>
        <div class={cardContentCls}>
          {$slots.default}
        </div>
      </div>
    );
  },
});

export default Card;
