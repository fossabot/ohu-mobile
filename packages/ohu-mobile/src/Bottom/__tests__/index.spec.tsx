import { mount } from '@vue/test-utils';
import Bottom from '..';
import Card from '../../Card';

describe('Bottom', () => {
  it('DOM structure', () => {
    const wrapper = mount(Bottom, {
      slots: {
        default: [
          Card
        ],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
