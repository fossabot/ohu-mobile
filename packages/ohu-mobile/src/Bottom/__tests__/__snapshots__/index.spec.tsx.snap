// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom DOM structure 1`] = `
<div
  class="ohu-bottom"
>
  <div
    class="ohu-card has-radius"
  >
    <div
      class="ohu-card__content"
    />
  </div>
</div>
`;
